lib/demos/material/divider_demo.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/gallery_localizations.dart';
+
+enum DividerDemoType {
+  horizontal,
+  vertical,
+}
+
+class DividerDemo extends StatelessWidget {
+  const DividerDemo({Key key, this.type}) : super(key: key);
+
+  final DividerDemoType type;
+
+  String _title(BuildContext context) {
+    switch (type) {
+      case DividerDemoType.horizontal:
+        return GalleryLocalizations.of(context).demoDividerTitle;
+      case DividerDemoType.vertical:
+        return GalleryLocalizations.of(context).demoVerticalDividerTitle;
+    }
+    return '';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget dividers;
+    switch (type) {
+      case DividerDemoType.horizontal:
+        dividers = _HorizontalDividerDemo();
+        break;
+      case DividerDemoType.vertical:
+        dividers = _VerticalDividerDemo();
+        break;
+    }
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(
+          _title(context),
+        ),
+      ),
+      body: dividers,
+    );
+  }
+}
+
+// BEGIN dividerDemo
+
+class _HorizontalDividerDemo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final localization = GalleryLocalizations.of(context);
+    return Container(
+      padding: const EdgeInsets.all(10),
+      child: Column(
+        children: [
+          Expanded(
+            child: _buildContainer(
+              text: localization.demoDividerAbove,
+              color: Colors.deepOrangeAccent,
+            ),
+          ),
+          const Divider(
+            color: Colors.grey,
+            height: 20,
+            thickness: 1,
+            indent: 20,
+            endIndent: 0,
+          ),
+          Expanded(
+            child: _buildContainer(
+              text: localization.demoDividerBelow,
+              color: Colors.deepPurpleAccent,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+// BEGIN verticalDividerDemo

```

Comment:
I think these tags are out of order, move this down to just above the `_VerticalDividerDemo` class

Reply:
I did it so that buildContainer code can be seen in both divider and vertical divider demo.

Reply:
oh thats cool, I didn't know it would work like that!